src/librustdoc/clean/mod.rs

```diff
@@ -1694,22 +1737,135 @@ impl<'a, 'tcx> Clean<Generics> for (&'a ty::Generics,
     }
 }
 
+// The point is to replace bounds with types.
+fn get_real_types(
+    generics: &Generics,
+    arg: &Type,
+    cx: &DocContext<'_, '_, '_>,
+) -> FxHashSet<Type> {
```

Comment:
If you use a `BTreeSet` instead, it comes pre-sorted - otherwise it comes out in an arbitrary order. I'm not sure if you wanted it to be properly sorted or just deduplicated, though.

Reply:
Dedup, the order doesn't matter. ;)